src/async.rs

```diff
@@ -0,0 +1,169 @@
+use crate::config::Response;
+use crate::error::{Error, ErrorResponse, RequestError};
+use crate::params::Headers;
+use futures::future::{self, Future};
+use futures::stream::Stream;
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+use reqwest::r#async::RequestBuilder;
+use serde::de::DeserializeOwned;
+
+#[derive(Clone)]
+pub struct Client {
+    client: reqwest::r#async::Client,
+    secret_key: String,
+    headers: Headers,
+}
+
+impl Client {
+    pub fn new<Str: Into<String>>(secret_key: Str) -> Client {
+        let client = reqwest::r#async::Client::new();
+        Client { client, secret_key: secret_key.into(), headers: Headers::default() }
+    }
+
+    /// Clones a new client with different headers.
+    ///
+    /// This is the recommended way to send requests for many different Stripe accounts
+    /// or with different Meta, Extra, and Expand headers while using the same secret key.
+    pub fn with_headers(&self, headers: Headers) -> Client {
+        let mut client = self.clone();
+        client.headers = headers;
+        client
+    }
+
+    /// Sets a value for the Stripe-Account header
+    ///
+    /// This is recommended if you are acting as only one Account for the lifetime of the client.
+    /// Otherwise, prefer `client.with(Headers{stripe_account: "acct_ABC", ..})`.
+    pub fn set_stripe_account<Str: Into<String>>(&mut self, account_id: Str) {
+        self.headers.stripe_account = Some(account_id.into());
+    }
+
+    /// Make a `GET` http request with just a path
+    pub fn get<T: DeserializeOwned + Send + 'static>(&self, path: &str) -> Response<T> {
+        let url = Client::url(path);
+        let request = self.client.get(&url).headers(self.headers());
+        send(request)
+    }
+
+    /// Make a `GET` http request with url query parameters
+    pub fn get_query<T: DeserializeOwned + Send + 'static, P: serde::Serialize>(
+        &self,
+        path: &str,
+        params: P,
+    ) -> Response<T> {
+        let url = match Client::url_with_params(path, params) {
```

Comment:
Oh `rust-try-future` is interesting!

I think I'll probably avoid the macro for now, just because it is used so few times, I don't want to have someone have to go find the non-standard macro definition (or ask external contributors to "fix" their PR if they don't use)